stan/math/prim/fun/to_int.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef STAN_MATH_PRIM_FUN_TO_INT_HPP
+#define STAN_MATH_PRIM_FUN_TO_INT_HPP
+
+#include <stan/math/prim/functor/apply_scalar_unary.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the input scalar as an integer type
+ *
+ * @tparam T type of argument (must be arithmetic)
+ * @param x argument
+ * @return iIteger value of argument
+ */
+template <typename T, require_arithmetic_t<T>* = nullptr>
+inline int to_int(T x) {
```

Comment:
[optional]
probably not worth specializing, but this is an inefficient implementation for `int` inputs because of the unnecessary tests